Implemented FAQ Section

ethernetvan opened the pull request:
Modified FAQ component to mimic the Stardew Valley community post board. Users can click on different categories to see questions about different topics (venue, general, food, items, lost & found, miscellaneous) and a Stardew Valley villager dialogue will appear, where users can click on specific questions.

Created faqData.ts to store all FAQ questions and responses. Contains dummy questions/answers like "Question1/Answer1" as placeholders for real questions.

<img width="691" height="366" alt="Screenshot 2025-10-10 at 11 31 16 PM" src="https://github.com/user-attachments/assets/ed71383c-83b1-4b97-aa7a-0641bea82ebb" />

ethernetvan committed:
Functional FAQ Dialogue

ethernetvan committed:
Switched To More Defined Colors

ethernetvan committed:
Redesign FAQ Board

ethernetvan committed:
Click Outside Dialogue Box to Close

sijun-kevin-hu reviewed (changes requested):
I really like the look of the FAQ!

Just a few changes:
1. Could you split the components and functions into multiple files for readability and so the FAQ file isn’t so compact?
2. For the Stardew Valley characters, I don’t think we can actually use the characters images and their names, so could you find something similar that isn’t copyrighted? If not, we can leave the images blank for now.

ethernetvan committed:
Removed Copyrighted Characters

ethernetvan committed:
Accessibility Improvements

- Changed some header levels
- Made interactive div a button
- Fixed React import issue
- Added Aria Labels & IDs

ethernetvan committed:
Separated Components to Different Files

Separated components and added additional comments

sijun-kevin-hu reviewed (approved):
GREAT

cnowdev reviewed (approved):
lgtm
